indexer/osm_editor.cpp

```diff
@@ -339,6 +339,22 @@ void Editor::ClearAllLocalEdits()
   Invalidate();
 }
 
+void Editor::OnMapDeregistered(platform::LocalCountryFile const & localFile)
+{
+  using FeaturePair = pair<MwmSet::MwmId, map<uint32_t, FeatureTypeInfo>>;
+  // Can not to find by MwmId because country already removed. So, find by country name.
+  auto matchedMwm =
+      find_if(begin(m_features), end(m_features), [&localFile](FeaturePair const & item) {
+        return item.first.GetInfo().get()->GetCountryName() == localFile.GetCountryName();
+      });
+
+  if (m_features.end() != matchedMwm)
+  {
+    m_features.erase(matchedMwm);
```

Comment:
Agree, this code is called each time mwm is deregistered, without any synchronization.